src/core/ext/filters/client_channel/xds/xds_bootstrap.cc

```diff
@@ -116,11 +140,11 @@ grpc_error* XdsBootstrap::ParseXdsServer(grpc_json* json) {
         error_list.push_back(GRPC_ERROR_CREATE_FROM_STATIC_STRING(
             "\"server_uri\" field is not a string"));
       }
-      if (server_uri_ != nullptr) {
+      if (server.server_uri != nullptr) {
         error_list.push_back(GRPC_ERROR_CREATE_FROM_STATIC_STRING(
             "duplicate \"server_uri\" field"));
       }
-      server_uri_ = child->value;
+      server.server_uri = child->value;
     } else if (strcmp(child->key, "channel_creds") == 0) {
       if (child->type != GRPC_JSON_ARRAY) {
         error_list.push_back(GRPC_ERROR_CREATE_FROM_STATIC_STRING(
```

Comment:
Ditto. If the type is not array, this server should probably be invalid.

Reply:
Or if null channel_creds is allowed, that should be fine.

Reply:
See above.  It doesn't matter, because the config parsing will fail as long as there's at least one error.